frontend/src/views/contract_template/upload-contract-template-button.tsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import { ActionButton } from "../../components/action-buttons";
+import { FaUpload } from "react-icons/fa";
+import { useDispatch } from "react-redux";
+import { uploadContractTemplate } from "../../api/actions";
+import {
+    Modal,
+    Row,
+    Col,
+    Container,
+    Form,
+    Button,
+    Spinner,
+} from "react-bootstrap";
+
+export function ConnectedUploadContractTemplateAction() {
+    const dispatch = useDispatch();
+    const [file, setFile] = React.useState<File | null>(null);
+    const [fileInputLabel, setFileInputLabel] = React.useState(
+        "Select an HTML template file."
+    );
+    const [dialogOpen, setDialogOpen] = React.useState<Boolean>(false);
+    const [inProgress, setInProgress] = React.useState(false);
+
+    async function _onFileChange(event: React.ChangeEvent<HTMLInputElement>) {
+        if (event?.target?.files) {
```

Comment:
I did NOT know you could do that. Since when?! 🤯

Reply:
It's a typescript feature :-)